helix-term/src/ui/picker/handlers.rs

```diff
@@ -137,17 +142,23 @@ impl<T: 'static + Send + Sync, D: 'static + Send + Sync> DynamicQueryHandler<T,
 }
 
 impl<T: 'static + Send + Sync, D: 'static + Send + Sync> AsyncHook for DynamicQueryHandler<T, D> {
-    type Event = Arc<str>;
+    type Event = DynamicQueryChange;
 
-    fn handle_event(&mut self, query: Self::Event, _timeout: Option<Instant>) -> Option<Instant> {
+    fn handle_event(&mut self, change: Self::Event, _timeout: Option<Instant>) -> Option<Instant> {
+        let DynamicQueryChange { query, is_paste } = change;
         if query == self.last_query {
             // If the search query reverts to the last one we requested, no need to
             // make a new request.
             self.query = None;
             None
         } else {
             self.query = Some(query);
-            Some(Instant::now() + self.debounce)
+            if is_paste {
```

Comment:
Sorry to bikeshed on naming, but I wonder if this should be something like `skip_debounce` instead, since you "[t]reat history inputs as pastes" (i.e. it's not _just_ if the event is a paste, but if the event should skip the debounce).

Reply:
actually I think it's better to change the comment in this case, inserting register contents is a paste

Reply:
Ah, okay! Then, that also works!